editor/src/components/navigator/navigator-item/component-picker.tsx

```diff
@@ -264,8 +313,145 @@ const FilterBar = React.memo((props: FilterBarProps) => {
   )
 })
 
+interface FilterButtonsProps {
+  components: Array<{
+    label: string
+    items: Array<InsertMenuItem>
+  }>
+}
+
+const FilterButtons = React.memo((props: FilterButtonsProps) => {
+  const { components } = props
+
+  const [focusedIndex, setFocusedIndex] = React.useState(0)
+
+  const setActiveIndexAll = React.useCallback(() => setFocusedIndex(0), [setFocusedIndex])
+
+  return (
+    <div
+      tabIndex={0}
+      aria-describedby='Component categories'
+      css={{
+        display: 'flex',
+        height: 30,
+        '&:not(:focus) button': {
+          backgroundColor: 'transparent',
+        },
+      }}
+      onKeyDown={(event) => {
+        if (event.key === 'ArrowRight') {
+          setFocusedIndex((prev) => Math.min(prev + 1, FILTER_CATEGORIES.length))
+        } else if (event.key === 'ArrowLeft') {
+          setFocusedIndex((prev) => Math.max(prev - 1, 0))
+        } else if (event.key === 'Enter') {
+          document
+            .getElementById(FILTER_CATEGORIES[Math.max(focusedIndex - 1, 0)])
+            ?.scrollIntoView({ block: 'start', behavior: 'smooth' })
+        } else {
+          return
+        }
```

Comment:
So keyboard navigation is a bit funky in this component and I am not sure why and what is going on behind the scenes with key events.

Basically, I think this popover container should have focus trapping and it doesn't. When I press tab to move focus, there are things in the background happening unrelated to the popover. Only things inside the popover should be affected until closed or Esc.

As a result, I have been focusing the `Filter...` text input and then pressing shift+tab to reverse focus into the category buttons.